lib/hooks/queries/useMarketSpotPrices.ts

```diff
@@ -25,58 +29,13 @@ export const useMarketSpotPrices = (marketId: number, blockNumber?: number) => {
   const query = useQuery(
     [id, assetPricesKey, pool, blockNumber],
     async () => {
-      if (isRpcSdk(sdk)) {
-        const spotPrices = new Map<number, Decimal>();
-        const outcomeWeights = pool.weights.filter(
-          (weight) =>
-            weight.assetId.toLocaleLowerCase() !==
-            pool.baseAsset.toLocaleLowerCase(),
-        );
-
-        if (market.status !== "Resolved") {
-          //base weight is equal to the sum of all other assets
-          const baseWeight = new Decimal(pool.totalWeight).div(2);
-
-          outcomeWeights.forEach((weight, index) => {
-            const spotPrice = calcSpotPrice(
-              basePoolBalance.toString(),
-              baseWeight,
-              balances[index].free.toString(),
-              weight.len,
-              0,
-            );
+      if (isRpcSdk(sdk) && isNA(basePoolBalance) === false) {
+        const spotPrices: MarketPrices =
+          market.status !== "Resolved"
+            ? calcMarketPrices(market, basePoolBalance as Decimal, balances)
```

Comment:
Don't think casting is needed here, the `isNA` check should have covered it but it doesn't seem to work